test/Test/Time/Units.hs

```diff
@@ -59,3 +60,21 @@ spec_Units = do
             floorUnit @Day (Time $ 5 :% 2) `shouldBe` 2
         it "returns 42ps when floor integer" $
             floorUnit (ps 42) `shouldBe` 42
+    describe "Formatting tests" $ do
+        it "4000 minutes should be formatted without ending-zeros" $
+            seriesF @'[DayUnit, HourUnit, MinuteUnit, SecondUnit] (minute 4000) `shouldBe` "2d18h40m"
+        it "4000 minutes should be formatted without beginning-zeros" $
+            seriesF @'[WeekUnit, DayUnit, HourUnit, MinuteUnit] (minute 4000) `shouldBe` "2d18h40m"
+        it "3601 sec should be formatted without middle-zeros" $
+            seriesF @'[HourUnit, MinuteUnit, SecondUnit] (sec 3601) `shouldBe` "1h1s"
+        it "works on rational nums" $
+            seriesF @'[HourUnit, SecondUnit, MillisecondUnit] (Time @MinuteUnit $ 3 :% 2) `shouldBe` "90s"
+        it "works without minutes formatting" $
+            seriesF @'[DayUnit, MinuteUnit, SecondUnit] (minute 4000) `shouldBe` "2d1120m"
+
+        it "4000 minutes should be formatted like 2d18h40m" $
```

Comment:
Let's also create test with formatting of all units. Like `fn 1 + week 1 + day 1 + ...`.
Of course, it's not possible to add  weeks and days in our library. But I think we can create some operator like this:

```haskell
(+:) :: Time a -> Time b -> Time b
```

I personally find it useful.